lib/compress/index.js

```diff
@@ -1598,6 +1599,52 @@ def_optimize(AST_Switch, function(self, compressor) {
     while (i < len) eliminate_branch(self.body[i++], body[body.length - 1]);
     self.body = body;
 
+    // group equivalent branches so they will be located next to each other,
+    // that way the next micro-optimization will merge them.
+    // ** bail micro-optimization if not a simple switch case with breaks
+    if (body.every(branch =>
+        (
+            branch instanceof AST_Default || // default
+            !branch.expression.has_side_effects(compressor) // or no_side_effects
+        )
+            && (
+                branch.body.length === 0 // empty
+                || is_break(branch.body[branch.body.length - 1], compressor) // or has a break
+                || is_return(branch.body[branch.body.length - 1]) // or has a return
+            )
+    )) {
+        for (let i = 0; i < body.length; i++) {
+            let branch = body[i];
+            if (branch.body.length === 0) continue;
+            if (!aborts(branch)) continue;
+            for (let j = i + 1; j < body.length; j++) {
+                let next = body[j];
+                if (next.body.length === 0) continue;
+                if (
+                    branches_equivalent(next, branch, false)
+                    || (j === body.length - 1 && branches_equivalent(next, branch, true))
```

Comment:
Nit: technically, since every branch aborts (per the `every` check), this second `branches_equivalent` isn't necessary (it can only be true if the last branch didn't abort).

Or, we could allow the final branch to omit a `break` and insert it if we need to move it. In which case, we'll need to keep this check.

Reply:
@jridgewell I think we should support "omitted break", most IDEs encourage you to remove an unnecessary `break`.
Will add support and a test cast

Reply:
@jridgewell  I've added support for this in my last commit 896ea1cd731a51d953e75a36b51013b5f769efc7